sw/device/lib/testing/test_framework/ottf_isrs.c

```diff
@@ -150,3 +150,9 @@ void ottf_external_isr(void) {
   generic_fault_print("External IRQ", ibex_mcause_read());
   abort();
 }
+
+OT_WEAK
+void ottf_internal_isr(void) {
```

Comment:
we would probably want to have the default one read the `mcause` register to determine whether we hit an internal or external NMI, similar to how the exception handler works above. See [here](https://ibex-core.readthedocs.io/en/latest/03_reference/exception_interrupts.html#internal-interrupts) for what values to check the `mcause` register for. Then you can create two sub functions (with OT_WEAK and OT_ALIAS) for both types of NMIs. So something like `ottf_load_integrity_error_handler()` and `ottf_external_nmi_handler()`. Does that make sense?